app/src/main/java/org/xbmc/kore/host/HostConnectionObserver.java

```diff
@@ -186,6 +211,10 @@ public void onSuccess(String result) {
                         checkWhatsPlaying();
                     }
 
+                    if (! playlistEventsObservers.isEmpty()) {
+                        checkPlaylist();
+                    }
+
```

Comment:
This is contrary to the current design. Through TCP we are not polling Kodi, we rely on Kodi's notifications to be notified that something changed, so we don't need to Poll anything (this runnable just pings Kodi, so that it doesn't close the TCP connection, which it does after 30 seconds of inactivity).
So, to keep consistency, we should implement the `Playlist.onAdd` and `Playlist.onRemove` notifications in `HostConnection`, catch them here, and call `checkPlaylist` there.

Reply:
It's there to make sure we also check if there are playlists available when we start the playlist fragment. If we would only depend on `onAdd`/`onRemove` then we would only see the available playlists when there is a change in one of the playlists on Kodi.
Another problem with `onAdd`/`onRemove` is that these are called for each item in a playlist when you add or remove multiple items. This makes it a bit harder, as we need to ignore subsequent calls. We could use a delay to prevent calling `checkPlaylist()` many times.

I'm currently thinking of refactoring this as follows:
* On registering a playlist observer call `checkPlaylist()` when using TCP.
* Implement `Playlist.onAdd` and `Playlist.onRemove` to handle changes to the playlist. Adding a delay (2 seconds?), to make sure when we delete a playlist with a trillion items we only call `checkPlaylist()` once after 2 seconds when we received the trillionth `onRemove` notification.

Reply:
- If you look a couple of lines up from this, you'll see that `checkWhatsPlaying` had the same exact problem, and the solution was calling it only when we were unintialized, not every time. That and a `forceReply` with last results, as done on the remote fragment.

- Use the same solution that was used on `checkWhatsPlaying`: use a static var to control whether a call is already in progress. Ignore race conditions, as they are inconsequential here.

Reply:
Code has been refactored and now uses onAdd/onRemove for TCP connections. I also implemented getting the playlist in the `tcpCheckerRunnable` only when there is a connection error. If I understand correctly that's the only situation in which the PlaylistFragment may keep showing an incorrect result when the connection is restored.

Reply:
Just for safety maybe we should also call `checkPlaylist` when `(hostState.lastCallResult == PlayerEventsObserver.PLAYER_NO_RESULT)`